lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -1352,6 +1434,125 @@ class ConvertAtenTransposeIntOp
 };
 } // namespace
 
+namespace {
+class ConvertAtenCatOp : public OpConversionPattern<AtenCatOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCatOp op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+    AtenCatOp::Adaptor adaptor(operands);
+
+    Value dimValue = op.dim();
+    int64_t dim;
+    if (!matchPattern(dimValue, m_TorchConstantInt(&dim)))
+      op.emitError("unimplemented: dim is not constant");
+
+    // Collect all the tensors to be concatenated.
+    auto tensorList = op.tensors();
+    auto listConstruct = tensorList.getDefiningOp<PrimListConstructOp>();
+    if (!listConstruct)
+      op.emitError("unimplemented: the tensor list is not from list construct");
+    auto tensors = llvm::makeArrayRef(llvm::to_vector<4>(
```

Comment:
makeArrayRef here is unsafe I think (lifetime issues). I think you can remove it anyway, because SmallVector converts to ArrayRef implicitly.